lib/elixir/lib/kernel.ex

```diff
@@ -5035,7 +5035,10 @@ defmodule Kernel do
   defmacro sigil_D(date_string, modifiers)
 
   defmacro sigil_D({:<<>>, _, [string]}, []) do
-    Macro.escape(Date.from_iso8601!(string))
+    {string, calendar} = Calendar.split_date_time_string(string, __CALLER__)
+    {year, month, day} = calendar.parse_date!(string)
+    {:ok, date} = Date.new(year, month, day, calendar)
```

Comment:
this results in:

```elixir
iex(1)> ~D[2019-01-0 Calendar.Holocene]
** (MatchError) no match of right hand side value: {:error, :invalid_date}
```

it would be good to have a similar error like here:

```elixir
iex(2)> ~D[2019-01-0]
** (ArgumentError) cannot parse "2019-01-0" as date, reason: :invalid_format
```